File: airbyte-commons-worker/src/main/java/io/airbyte/workers/context/ReplicationFeatureFlags.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.workers.context;

/**
 * Feature flags to consider during a Replication job.
 */
public record ReplicationFeatureFlags(boolean isDestinationTimeoutEnabled,
                                      int workloadHeartbeatRate,
                                      long workloadHeartbeatTimeoutInMinutes,
                                      boolean failOnInvalidChecksum,
                                      boolean logStateMsgs,
                                      boolean logConnectorMsgs) {}

File: airbyte-worker-models/src/main/kotlin/io/airbyte/workers/models/ReplicationFeatureFlags.kt
/*
 * Copyright (c) 2020-2026 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.workers.models

import io.airbyte.featureflag.Flag

data class ReplicationFeatureFlags(
  val featureFlags: List<Flag<*>>,
)
